File: Python/Program/RGBController.py
class RGBController:
    def __init__(self, warna="Rainbow", bright_level=50, is_on=False):
        self.__warna = warna
        self.__bright_level = bright_level
        self.__is_on = is_on
    
    def set_warna(self, warna):
        self.__warna = warna
    
    def set_bright_level(self, bright_level):
        if 0 <= bright_level <= 100:
            self.__bright_level = bright_level
    
    def set_is_on(self, is_on):
        self.__is_on = is_on
    
    def get_warna(self):
        return self.__warna
    
    def get_bright_level(self):
        return self.__bright_level
    
    def get_is_on(self):
        return self.__is_on
    
    def toggle_rgb(self):
        self.__is_on = not self.__is_on
        print(f"RGB {'diaktifkan' if self.__is_on else 'dinonaktifkan'}")

File: CPP/Program/RGBController.cpp
#ifndef RGBCONTROLLER_CPP
#define RGBCONTROLLER_CPP

#include <iostream>
#include <string>
using namespace std;

class RGBController {
private:
    string warna;
    int brightLevel;
    bool isOn;
    
public:
    RGBController() {
        this->isOn = false;
        this->brightLevel = 50;
        this->warna = "Rainbow";
    }
    
    RGBController(string warna, int brightLevel, bool isOn) {
        this->warna = warna;
        this->brightLevel = brightLevel;
        this->isOn = isOn;
    }
    
    void setWarna(string warna) {
        this->warna = warna;
    }
    
    void setBrightLevel(int brightLevel) {
        if (brightLevel >= 0 && brightLevel <= 100) {
            this->brightLevel = brightLevel;
        }
    }
    
    void setIsOn(bool isOn) {
        this->isOn = isOn;
    }
    
    string getWarna() {
        return this->warna;
    }
    
    int getBrightLevel() {
        return this->brightLevel;
    }
    
    bool getIsOn() {
        return this->isOn;
    }
    
    void toggleRGB() {
        this->isOn = !this->isOn;
        cout << "RGB " << (this->isOn ? "diaktifkan" : "dinonaktifkan") << endl;
    }
    
    ~RGBController() {
    }
};

#endif // RGBCONTROLLER_CPP